### PR TITLE
fix: [CDS-53837]: made text changes and more coverage

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributo

Pipelines:
Screen Shot 2023-03-07 at 12 24 33 PM
r license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

Runtime:
Screen Shot 2023-03-07 at 12 24 22 PM

Templates:
Screen Shot 2023-03-07 at 12 25 47 PM

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger jest

**Review by cjlee01 — APPROVED** [review]

File: src/modules/20-rbac/components/UserGroupsInput/UserGroupExpressionInput.tsx [hunk 1]
```diff
@@ -5,49 +5,76 @@
...
+  const userGroupExpression = get(formik.values, 'userGroupExpression', '')
   const { getString } = useStrings()
+
+  useEffect(() => {
+    if (value && Array.isArray(value)) {
+      formik.setFieldValue('userGroupExpression', ExpressionTypes.Single)
+    } else if (isValueExpression(value)) {
+      formik.setFieldValue('userGroupExpression', ExpressionTypes.Combined)
+    }
+  }, [])
```
Comment (rameshx) [inline src/modules/20-rbac/components/UserGroupsInput/UserGroupExpressionInput.tsx:38]: I'm guessing this effect is used for setting the initial value of `userGroupExpression`, can this logic be moved to the place where formik's initial values are passed?

The issue is that dependencies used inside the effect are not added in line 38.

**Review by tan-nhu — APPROVED** [review]

File: src/modules/20-rbac/components/UserGroupsInput/UserGroupExpressionInput.tsx (hunk 1 above)
Comment (sahithikolichala) [reply src/modules/20-rbac/components/UserGroupsInput/UserGroupExpressionInput.tsx:38]: this need not be done at multiple places 
This is a common component which is shared between (pipeline, runtime and templates)

Comment (sahithikolichala) [reply src/modules/20-rbac/components/UserGroupsInput/UserGroupExpressionInput.tsx:38]: i want it to be done  only once ,i.e., at component load time

**Review by cjlee01 — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Comment by sahithikolichala** [comment]
retrigger SorryCypressCV-Cypress_Pipeline

**Comment by sahithikolichala** [comment]
retrigger SorryCypress-Pipeline

**Comment by sahithikolichala** [comment]
retrigger SorryCypressPipeline-Cypress_Pipeline

**Comment by vkbansal-harness** [comment]
retrigger cypress-cv

**Comment by vkbansal-harness** [comment]
retrigger cypress-pipeline

**Pull request merged**